modules/javafx.graphics/src/main/native-font/directwrite.cpp

```diff
@@ -866,10 +873,20 @@ JNIEXPORT jlong JNICALL OS_NATIVE(_1WICCreateImagingFactory)
             CLSCTX_INPROC_SERVER,
             IID_PPV_ARGS(&result)
             );
+    if (!SUCCEEDED(hr)) goto fail;
 
-    /* Unload COM as no other COM objects will be create directly */
+    /* Setup a fibre local storage slot to call WICImagingFactoryFlsCallback()
+     * with the created IWICImagingFactory on thread exit. */
+    DWORD flsIndex = FlsAlloc(WICImagingFactoryFlsCallback);
+    if (flsIndex == FLS_OUT_OF_INDEXES) goto fail;
+    FlsSetValue(flsIndex, result);
```

Comment:
I think you can skip this call, given that there is no need to use it to call Release in the callback method.